internal/kafka/internal/services/quota/quota_management_list_service.go

```diff
@@ -161,6 +170,14 @@ func (q QuotaManagementListService) ReserveQuota(kafka *dbapi.KafkaRequest) (str
 // 7) if [6] fails, return the first defined billing model
 func (q QuotaManagementListService) detectBillingModel(kafka *dbapi.KafkaRequest) (string, *errors.ServiceError) {
 	if kafka.DesiredKafkaBillingModel != "" {
+		instanceType, err := q.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(kafka.InstanceType)
```

Comment:
Could you explain why is this added?

Reply:
because without that any `DesiredBillingModel` was valid, even if not present into the configuration